internal/utils/maps.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import "fmt"
+
+// Merge maps recursively merges the values of b into a copy of a, preferring the values from b
+func MergeMaps(a, b map[string]interface{}) map[string]interface{} {
+	out := make(map[string]interface{}, len(a))
+	for k, v := range a {
+		out[k] = v
+	}
+	for k, v := range b {
+		if v, ok := v.(map[string]interface{}); ok {
+			if bv, ok := out[k]; ok {
+				if bv, ok := bv.(map[string]interface{}); ok {
+					out[k] = MergeMaps(bv, v)
+					continue
+				}
+			}
+		}
+		out[k] = v
+	}
+	return out
+}
+
+// GetPath gets a value by a path in a nested map[string]interface{}.
+// An error is returned if:
+//   - At any point along the path, the map is not a map[string]interface{}
+//   - At any point along the path, the path segment does not appear in the map
+func GetPath(m map[string]interface{}, path []string) (interface{}, error) {
```

Comment:
Check out https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured, it might be useful

Reply:
Implemented :+1: